File: sources/javascript/06_char_count/main.js
// task 06 char_count — expected output: 10000000
// build: none    run: node main.js | bun main.js | deno run main.js

const text = 'abcdefghij'.repeat(10000000);
let count = 0;
for (let i = 0; i < text.length; i++) {
  const ch = text.charCodeAt(i);
  if (ch === 97) continue; // 'a'
  else if (ch === 101) continue; // 'e'
  else if (ch === 104) count += 1; // 'h'
}
console.log(count);

File: sources/rust/06_char_count/main.rs
// task 06 char_count — expected output: 10000000
// build: rustc -O -o prog main.rs    run: ./prog

fn main() {
    // One bulk repeat: 100_000_000 characters, built before the scan.
    let text = "abcdefghij".repeat(10_000_000);

    let mut count: u64 = 0;
    for &ch in text.as_bytes() {
        if ch == b'a' {
            continue;
        } else if ch == b'e' {
            continue;
        } else if ch == b'h' {
            count += 1;
        } else {
            continue;
        }
    }

    println!("{}", count);
}
